messenger/src/redis_messenger.rs

```diff
@@ -77,7 +83,8 @@ impl Messenger for RedisMessenger {
         }
     }
 
-    fn send(&mut self, stream_key: &'static str, bytes: &[u8]) -> Result<()> {
+    //impl Future<Output = Result<()>> + 'a {
+    async fn send(&mut self, stream_key: &'static str, bytes: &[u8]) -> Result<()> {
         // Check if stream is configured.
```

Comment:
When streaming time series data, we need to think carefully about the order of messages coming through each step of the pipeline. If messages are guaranteed to be delivered in order of time, the write operations in the database may have significant performance improvements. This may or may not be the case for our current data store.  But, if things get out of order, then the database may have to re-order the datapoints itself, which could be problematic. 
Again this is theoretical, but here's an example. Async message publication to Redis results in out-of-order time series data in the stream. The consumer app consumes the messages sequentially from the stream and publishes write operations in small batches. If the data is unordered, the database may have to sort before writing to disk. 
I'm not sure that anything is problematic in this case, but this is the kind of thing we should be thinking about.

Reply:
Talked to @austbot and you offline about this.  I think we will plan to use a sequence number on the consumer side to deal with this for this project, and it will be fine for this specific case, but I definitely see the more generic use case being applicable.